src/Api/Controllers/ProviderUsersController.cs

```diff
@@ -67,8 +67,14 @@ public async Task Invite(Guid providerId, [FromBody]ProviderUserInviteRequestMod
                 throw new NotFoundException();
             }
 
-            var userId = _userService.GetProperUserId(User);
-            await _providerService.InviteUserAsync(providerId, userId.Value, new ProviderUserInvite(model));
+            var invite = new ProviderUserInvite<string>
```

Comment:
I think I'd prefer the `ProviderUserInvite` class to handle this and the reinvite constructions. I think because of the generic type it'll have to be static factory methods, but I'm not totally sure on that. Would prefer constructors if possible.

Reply:
Factory methods only, but implemented as suggested